KILTprotocol/kilt-node
# chore: clippy lints

Fixes https://github.com/KILTprotocol/ticket/issues/3325.

Add some basic lints (can always be expanded for either runtime-specific code or all code via the configurable rustc flags in the new config file). I updated the code to address all places in which the new lints generate an error, including all weight files and weight templates. I also updated the CI check step to check for the specific WASM-targeted, no-std code that will be part of our runtime.

## How to test

To check runtime code against the new lint, just run the same command as in the check workflow file:

`cargo clippy --locked --target wasm32-unknown-unknown --no-default-features --workspace --exclude kilt-parachain --exclude standalone-node --exclude xcm-integration-tests --exclude 'dip-provider*' --exclude 'dip-consumer*'`
